src/components/accessRequests/accessRequests.tsx

```diff
@@ -76,26 +115,32 @@ const AccessRequests = () => {
           });
         }
       }
-      setRequests(requests);
+      if (accessRequests !== null) {
+        setRequests(accessRequests);
+        if (!filteredRequests)
+          setFilteredRequests(
+            accessRequests.filter((x) => x.status === "pending")
+          );
+      }
     }
-    fetchData();
-  }, [showMessage, user]);
+    if (requests === null || requests === undefined) fetchData();
+  }, [filteredRequests, requests, showMessage, user]);
```

Comment:
Remove filteredRequests and requests here as dependencies, then this function is only run once as it should be.

Reply:
Done